File: Java-CodeServer/src/main/resources/js/prompto/csharp/CSharpThisExpression.js
var CSharpExpression = require("./CSharpExpression").CSharpExpression;
var ThisExpression = require("../expression/ThisExpression").ThisExpression;

function CSharpThisExpression() {
    CSharpExpression.call(this);
	this.expression = new ThisExpression();
	return this;
}

CSharpThisExpression.prototype = Object.create(CSharpExpression.prototype);
CSharpThisExpression.prototype.constructor = CSharpThisExpression;

CSharpThisExpression.prototype.toDialect = function(writer) {
    return this.expression.toDialect(writer);
};

CSharpThisExpression.prototype.toString = function() {
	return this.expression.toString();
};

exports.CSharpThisExpression = CSharpThisExpression;

File: Core/src/main/java/prompto/csharp/CSharpThisExpression.java
package prompto.csharp;

import prompto.expression.ThisExpression;
import prompto.utils.CodeWriter;

public class CSharpThisExpression implements CSharpExpression {
	
	ThisExpression expression = new ThisExpression();
	
	@Override
	public void toDialect(CodeWriter writer) {
		expression.toDialect(writer);
	}
}
